### ISSUE TITLE
Handler name in serverless.template is incorrect after generation(short issue description)

### ISSUE THREAD

**Issue by SogdianDotNet** [aws/aws-lambda-dotnet#1624]
### Describe the bug

Function handler name incorrectly generated. I am using `**Amazon.Lambda.Annotations**` **v.1.1.0**

This is blocking for me as I can't `debug` it locally nor can I use this for production.

I am using different names here for my namespace and project name.

Projectname: LambdaBamba
AssemblyName: LambdaBamba
Function class: Function.cs
Function method: FunctionHandler

### Expected Behavior

I expect the handlername in my serverless.template to be generated as:

```
"ImageConfig": {
  "Command": [
    "LambdaBamba::LambdaBamba.Function_Generated::FunctionHandler"
  ]
}
```

### Current Behavior

But what I get is:

```
"ImageConfig": {
  "Command": [
    "LambdaBamba"
  ]
}
```

### Reproduction Steps

I created a new AWS serverless lambda project in Visual Studio using the **Custom Runtime Function** blueprint template (`.NET 8`).

I followed this video of James Eastham [youtube](https://www.youtube.com/watch?v=kyb16r-Oul0).

I am also using the lambda annotations (LambdaStartup, LambdaFunction).

### AWS .NET SDK and/or Package version used

AWSSDK 3.7.300.1

### Targeted .NET Platform

.NET 8

### Operating System and version

Windows 10

**Comment by ashishdhingra** [comment]
@SogdianDotNet Good afternoon. Thanks for opening the issue. For custom runtime functions, the name of the handler is the executable assembly itself (default `bootstrap`). In your case it is `LambdaBamba` assuming that it is set as `AssemblyName` for your project. Do you have `function-handler` defined in `aws-lambda-tools-defaults.json` by any chance?

Thanks,
Ashish

**Comment by SogdianDotNet** [comment]
> @SogdianDotNet Good afternoon. Thanks for opening the issue. For custom runtime functions, the name of the handler is the executable assembly itself (default `bootstrap`). In your case it is `LambdaBamba` assuming that it is set as `AssemblyName` for your project.
> 
> * Do you have `function-handler` defined in `aws-lambda-tools-defaults.json` by any chance?
> * when i use Visual Studio `Custom Runtime Function blueprint template (.NET 8)`, it only creates a default function handler with no `serverless.template`. Could you share minimal code solution for reproducing the issue?
> 
> Thanks, Ashish

@ashishdhingra 

Yes, and with testing this locally, I mean using the Lambda Mock tool.

- I do have `function-handler` defined in `aws-lambda-tools-default.json`

**Function.cs:**
```
public class Function
{
    private readonly ISomeRepository _someRepo;
    private readonly ISomeRepository1 _someRepo1;

    public Function(ISomeRepository _someRepo, _someRepo1)
    {
        ArgumentNullException.ThrowIfNull(someRepo);
        ArgumentNullException.ThrowIfNull(someRepo1);
        _someRepo= someRepo;
        _someRepo1 = someRepo1;
    }

    [LambdaFunction(PackageType = LambdaPackageType.Image, Timeout = 10)]
    public async Task FunctionHandler(string input, ILambdaContext context)
    {
        // some code
    }
}
```

**AssemblyInfo.cs**
```
using Amazon.Lambda.Annotations;
using Amazon.Lambda.Core;

[assembly: LambdaSerializer(typeof(Amazon.Lambda.Serialization.SystemTextJson.DefaultLambdaJsonSerializer))]
[assembly: LambdaGlobalProperties(GenerateMain = true, Runtime = "provided.al2")]
```

**aws-lambda-tools-default.json**
`{
  "Information": [
    "This file provides default values for the deployment wizard inside Visual Studio and the AWS Lambda commands added to the .NET Core CLI.",
    "To learn more about the Lambda commands with the .NET Core CLI execute the following command at the command line in the project root directory.",
    "dotnet lambda help",
    "All the command line options for the Lambda command can be specified in this file."
  ],
  "profile": "user-deploy",
  "region": "eu-west-1",
  "configuration": "Release",
  "function-architecture": "x86_64",
  "function-runtime": "provided.al2",
  "function-memory-size": 256,
  "function-timeout": 10,
  "function-handler": "bootstrap",
  "msbuild-parameters": "--self-contained true",
  "s3-bucket": "somebucket",
  "template": "serverless.template",
  "stack-name": "somebucket"
}`

**serverless.template**
`{
  "AWSTemplateFormatVersion": "2010-09-09",
  "Transform": "AWS::Serverless-2016-10-31",
  "Description": "This template is partially managed by Amazon.Lambda.Annotations (v1.0.0.0).",
  "Resources": {
    "TajikMartyrsLambdaCacheHomePageTextsFunctionFunctionHandlerGenerated": {
      "Type": "AWS::Serverless::Function",
      "Metadata": {
        "Tool": "Amazon.Lambda.Annotations"
      },
      "Properties": {
        "MemorySize": 256,
        "Timeout": 10,
        "Policies": [
          "AWSLambdaBasicExecutionRole"
        ],
        "PackageType": "Image",
        "Environment": {
          "Variables": {
            "ANNOTATIONS_HANDLER": "FunctionHandler"
          }
        },
        "ImageUri": ".",
        "ImageConfig": {
          "Command": [
            "LambdaBamba"
          ]
        }
      }
    }
  }
}`

When I run the lambda locally, this is what I see in the console log:

```
... Using function handler LambdaBamba
... Using payload with the value payload.json
... Setting AWS_PROFILE environment variable to user-deploy.
... Setting AWS_REGION environment variable to eu-west-1.
Captured Log information:

Request failed to execute
Error:
Invalid format for function handler string LambdaBamba. Format is <assembly>::<type-name>::<method>.
```

**Comment by normj** [comment]
What James is showing in his video is how to deploy your Lambda function as an executable assembly. To use the test tool you need to use the executable assembly page in the tool. The page has some instructions and here is some from the repo's README https://github.com/aws/aws-lambda-dotnet/tree/master/Tools/LambdaTestTool#testing-executable-assemblies

Basically for executable assemblies the tool tool acts as a remote endpoint and then you launch your executable assembly with the `AWS_LAMBDA_RUNTIME_API` environment variable configured to the endpoint in the test tool. Then the executable assembly will poll for messages from the test tool and you can queue up events to send to it.

**Comment by ashishdhingra** [comment]
> What James is showing in his video is how to deploy your Lambda function as an executable assembly. To use the test tool you need to use the executable assembly page in the tool. The page has some instructions and here is some from the repo's README https://github.com/aws/aws-lambda-dotnet/tree/master/Tools/LambdaTestTool#testing-executable-assemblies
> 
> Basically for executable assemblies the tool tool acts as a remote endpoint and then you launch your executable assembly with the `AWS_LAMBDA_RUNTIME_API` environment variable configured to the endpoint in the test tool. Then the executable assembly will poll for messages from the test tool and you can queue up events to send to it.

@SogdianDotNet Please advise if guidance provided by @normj works for you.
